### DISCUSSION TITLE
Instructions for latest wxmaxima release?

### DISCUSSION THREAD

**Discussion by mauriva** [wxMaxima-developers/wxmaxima#1478] [Q&A]
I am using the latest version of Maxima (5.44) on Windows 10, which contains wxMaxima at version 20.06.6. Looking for updates, I am notified of the existence of version 21.01.0 of wxmaxima.
Unfortunately I have nowhere found a compiled version of wxmaxima, nor any instructions to do it myself.
Why report the update if it cannot be installed for the end user?

**Comment by Parlazo** [comment]
Same here... but i think there are no major changes on windows.. so its hopefully not so important to change any thing if it is running.

**Comment by daute** [comment]
Sorry for the late answer. For the recent releases a Windows installer (cross-compiled on Linux) is provided (containing just wxMaxima, so you can use a recent wxMaxima along with the current Maxima installer (I hope a new Maxima release (containing of course the current wxMaxima) will occur in the near future)).

On
https://wolfgang.dautermann.at/maxima/nightlybuild/
I provide nightly builds (Windows installers, crosscompiled on Linux) of the current Maxima + wxMaxima development sources.

Best regards, Wolfgang

**Reply by Parlazo** [reply]
May you can tell...
I have wxMaxima an Maxima and GnuPlot all included installed...
is it now possible to get the wxMaxima only installer and just run it over the old wxMaxima with out loosing any configuration in wxMaxima and will it reuse the Maxima and GnuPlot settings?
on windows?

**Reply by daute** [reply]
You get a second installation (by default Maxima installs in `C:\maxima-5.44.0` (and there you also find wxMaxima (20.06.6) and also an entry in the start menu (under Maxima-5.44.0). This wxMaxima installer installs by default in `C:\wxMaxima-21.04.0` and you get another entry in the start menu. [I do not like the standard hierarchy, cause in "`C:\Program files`" is a space, which often causes problems and must be escaped. No idea, how Microsoft got THAT idea for the naming of the directory...]

So you can install (and execute) both versions at the same time. It seems not all wxMaxima configuration options are reused, maybe I can look at that issue on the weekend...

Best regards, Wolfgang

**Reply by Parlazo** [reply]
may there is a way to use the windows environment variables
https://ss64.com/nt/syntax-variables.html
in windows file explorer this %ProgramFiles% will jump directly into the "C:\Program Files" folder.
i dont know
there also exists variables as %ProgramData% or %USERNAME% to assemble some path magic

**Reply by daute** [reply]
No, that does not help. The idea of using a space in system directories is just *very* strange. 

This command:
```
C:\> mkdir %ProgramFiles%\aaa
```

creates 3 directories:
```
C:\Program
C:\Files
C:\Files\aaa
```

Not
`
C:\Program Files\aaa
`
as one would expect (admin permissions are required to create that directory). One still have to care about the strange space character.

I stick to my opinion: a **very** strange (bad) idea...

**Reply by Parlazo** [reply]
`
md "%SystemDrive%\Users\%USERNAME%\wxMaxima Test Gen Delete ME Dir"
`

will generate from CMD (the console) `C:\Users\usernamedir\wxMaxima Test Gen Delete ME Dir`

`MD` is case insensitive and means some thing like "make directory"

Windows does need path surrounded by quotes if there are spaces in any folder or file name, but at least at the beginning of a path string and then it will guess the missing quote at the end of sting, how ever quoted path does work also if there is no space in the string.

**Reply by Parlazo** [reply]
additional may important in the windows registry if used
`key_name = "C:\\Program Files\\blah\\file.ext"`
backslash must be doubled to escape them to be working

**Reply by daute** [reply]
Yes, I know, it can be escaped by using quotes. But I create not only the wxMaxima installer, but also the Windows installer for Maxima, where I depend on many other free software packages (Clisp, SBCL, ABCL, Tcl/Tk, Gnuplot, VTK, ...), where I am *not* a developer and can not (easy) solve bugs with such issues.

And using an installation directory, which does not contain 'problematic' characters, just causes **less headache**. Therefore I am using it. I am doing that in my spare time, not as a paid programmer.

Best regards, Wolfgang

**Reply by Parlazo** [reply]
yes this is true ... some applications have problems with space in a path on windows
but the best argument is to avoid the admin privileges
a application which does not need such privileges is some how very handy... it may even could work as a portable app... specially for students this is a good option if they have to work on different computers at different places.
i like this idea too

**Reply by mauriva** [reply]
> 
> 
> May you can tell...
> I have wxMaxima an Maxima and GnuPlot all included installed...
> is it now possible to get the wxMaxima only installer and just run it over the old wxMaxima with out loosing any configuration in wxMaxima and will it reuse the Maxima and GnuPlot settings?
> on windows?

This is what I'm looking for...
I confess that I am not interested in having two different installations of wxMaxima: I would like the installer to overwrite the old version.
With the installer currently available on GitHub I don't know if this is possible ...

**Reply by Parlazo** [reply]
> **less headache**

So windows space path as `C:\Program Files\aaa` is not supported.
The user has to install wxMaxima with the related applications into a non spaced folder.
Maybe `C:\wxMaxima` or `C:\DOSapps\wxMaxima`, any where as long it contains no space in the path.
All related applications as gnuplot or maxima are in a subfolder of wxMaxima.
This, basicaly, looks very dynamic, similar to a portable app on a USB stick,
The only problem which avoid wxMaxima from being portable on a USB stick is, the settings in the windows registry.

In the old forum many people ask about a settings file, so they can manipulate it manually or use different setting files, some times to get hand on settings which are not covered by the configuration dialog.

I think in such a situation it would also prevent head ache if the settings would just be in a *.ini file at the side of the *.exe file.
Most questions about the settings file would be answered in short, most users may even would find the file by them self.
Basically wxMaxima would be relative independent of permanent Windows structures, as the `registry` or` c:\PROGRA~1` or `c:\PROGRA~2`.
And this would make wxMaxima perfectly portable on a USB stick by it self without `PortableApps.com` tricks.
I think this would be a perfect feature and not only serve the user but also wxMaxima it self.
Students could easily take they´r own personal wxMaxima to the library or to school or to they´r best friends or even use it in they´r (home-)office. People close to them would ask, what strange app they are using, and so wxMaxima would get more attention, and this would maybe build up to more contributions in any form.

Just an idea... thanks any way :100:

**Reply by zmth** [reply]
yea that would be much better to keep everything out of the registry. That damn microsoft puts out such crappy software.For example like most everyone else I am the one and only person who is ever going to be using this computer or anything so i don't want any crap about users or administrator/etc. And what's more than that is that damn microsoft has the nerve to tell me i don't have permission to do something. Anyone who makes any software like that has got to be the dumbest jerk ever ! Because of this issue and many others i was thinking of transferring to Linux but i know see those jerks have also screwed things up with Users and worthless junk like that!

**Reply by gunterkoenigsmann** [reply]
The problem with the registry is that Microsoft insists that windows software had to store its settings there, except of large boobs of data (which wxMaxima doesn't want to store in the config storage, anyway.)

**Reply by gunterkoenigsmann** [reply]
...but actually you can tell your windows to pass wxMaxima command-line-switches on startup. There is one that makes it store its config in s file.
